qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -91,6 +91,17 @@ def __init__(
         self.initial_state = initial_state
         self.callback = callback
         self._kwargs = kwargs
+        self._vqe = VQE(
+            ansatz=ansatz,
```

Comment:
Is this the line of culprit which you were referring to above when you said you had problems with `UCC`?
I actually do not see a reason for setting ansatz already up here. You can simply leave it at None and only set it during `get_solver` as in the past.

Essentially the idea here should be, that all the attributes which are handled by the factory (`ansatz` and in the future `initial_point` #571), are set during `get_solver` on some internal minimum eigensolver instance. The remaining property setters should be deprecated in favor of exposing the `MinimumEigensolver` directly (I think that's what you attempted in the numpy factory above, but I do not see you exposing the `self._minimum_eigensolver` via a property getter which is required for the user to actually have access to it.

Reply:
@mrossinek You are right! Thank you for the comment!